File: exercise/php/day13/tests/FakeCatalog.php
<?php

namespace Tests;

use SantaMarket\Product;
use SantaMarket\SantamarketCatalog;

class FakeCatalog implements SantamarketCatalog
{
    private array $products = [];
    private array $prices = [];

    public function addProduct(Product $product, float $price): void
    {
        $this->products[$product->getName()] = $product;
        $this->prices[$product->getName()] = $price;
    }

    public function getUnitPrice(Product $product): float
    {
        return $this->prices[$product->getName()];
    }
}

File: exercise/java/day13/src/test/java/santamarket/model/FakeCatalog.java
package santamarket.model;

import java.util.HashMap;
import java.util.Map;

public class FakeCatalog implements SantamarketCatalog {
    private final Map<String, Product> products = new HashMap<>();
    private final Map<String, Double> prices = new HashMap<>();

    @Override
    public void addProduct(Product product, double price) {
        this.products.put(product.getName(), product);
        this.prices.put(product.getName(), price);
    }

    @Override
    public double getUnitPrice(Product p) {
        return this.prices.get(p.getName());
    }
}
